environment/src/main/java/com/sequenceiq/environment/exception/mapper/SearchCauseExceptionMapper.java

```diff
@@ -12,13 +14,29 @@
     @Override
     Response.Status getResponseStatus(T exception) {
         Response.Status defaultResponse = Response.Status.BAD_REQUEST;
-        if (exception.getCause() != null && Exception.class == exception.getCause().getClass()) {
+        if (exception == null || exception.getCause() != null && Exception.class == exception.getCause().getClass()) {
             return defaultResponse;
         }
-        BaseExceptionMapper exceptionMapper = (BaseExceptionMapper<? extends Throwable>) providers.getExceptionMapper(exception.getCause().getClass());
-        if (exceptionMapper == null) {
+        Pair<BaseExceptionMapper, ? extends Throwable> pair = searchRecursively(exception);
+        if (pair.getKey() == null || pair.getKey() == this) {
             return defaultResponse;
         }
-        return exceptionMapper.getResponseStatus(exception.getCause());
+        return pair.getKey().getResponseStatus(pair.getValue());
+    }
+
+    private Pair<BaseExceptionMapper, Throwable> searchRecursively(Throwable exception) {
+        if (exception.getCause() == null) {
+            BaseExceptionMapper exceptionMapper = getExceptionMapper(exception);
+            return Pair.of(exceptionMapper, exception);
+        }
+        BaseExceptionMapper<? extends Throwable> exceptionMapper = getExceptionMapper(exception.getCause());
+        if (exceptionMapper == null || exceptionMapper instanceof DefaultExceptionMapper) {
+            return searchRecursively(exception.getCause());
+        }
+        return Pair.of(exceptionMapper, exception.getCause());
```

Comment:
The same for this method too.

Reply:
Do you want to merge some return statements or I should hold the result in a variable and return with them in one place?

Reply:
one global variable for the return value, of course the recursive call could remain as it isnow